File: library/src/main/java/com/nshmura/strictmodenotifier/StrictModeNotifierSingleThreadFactory.java
package com.nshmura.strictmodenotifier;

import java.util.concurrent.ThreadFactory;

/**
 * This is intended to only be used with a single thread executor.
 */
final class StrictModeNotifierSingleThreadFactory implements ThreadFactory {

  private final String threadName;

  StrictModeNotifierSingleThreadFactory(String threadName) {
    this.threadName = "StrictModeNotifier-" + threadName;
  }

  @Override public Thread newThread(Runnable runnable) {
    return new Thread(runnable, threadName);
  }
}

File: notifier/src/main/java/io/github/juanimoli/strictmode/notifier/StrictModeNotifierSingleThreadFactory.kt
package io.github.juanimoli.strictmode.notifier

import java.util.concurrent.ThreadFactory

/**
 * This is intended to only be used with a single thread executor.
 */
internal class StrictModeNotifierSingleThreadFactory(threadName: String) : ThreadFactory {
    private val threadName: String = "StrictModeNotifier-$threadName"

    override fun newThread(runnable: Runnable): Thread {
        return Thread(runnable, threadName)
    }
}
